File: Data Structures & Algorithms/duplicate-integer/submission-2.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        int n=nums.length;
        for(int i=0; i<n; i++){
            for(int j=0; j<n; j++)
                if(nums[i]==nums[j] && j!=i) return true;
        }
        return false;

    }   //
}

File: Data Structures & Algorithms/contains-duplicate-ii/submission-2.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        mp = {} # number -> index

        for i, n in enumerate(nums):
            if n in mp and i - mp[n] <= k:
                    return True
            mp[n] = i
        return False